### PR TITLE
Replace adhoc yaml with serde

### PR DESCRIPTION
**Pull request by Ichmed**
first part of https://github.com/Elektrobit/flake-pilot/issues/128

Includes only podman config, will copy the approach for firecracker and/or create generic version for all back ends if approved

Adds lazy_static for config initialization, can be removed if rust version is bumped to one that includes [OnceCell](https://doc.rust-lang.org/stable/std/cell/struct.OnceCell.html)

### CODE REVIEW

**Comment by Ichmed** [comment]
I had to adjust the config slightly because apparently `rust_yaml` and `serde_yaml` treat the ~ slightly differently (this version wanted to use the user "~" instead of the default user)

**Comment by schaefi** [comment]
Great this is a good pull request to review and test, thanks much :+1: I will take a closer look and test at the end of the week. Until then my plate is full with other stuff. On a first look it really looks good, again thanks much appreciated

**Review by schaefi — CHANGES_REQUESTED** [review]
I tested the change and it works for me but I think we are loosing an important feature, see my comments. Thanks

File: podman-pilot/src/main.rs [hunk 1]
```diff
@@ -32,19 +32,18 @@ use env_logger::Env;
...
 pub mod podman;
 pub mod defaults;
+pub mod config;
 
 fn main() {
     setup_logger();
 
     let program_path = app_path::program_abs_path();
     let program_name = app_path::basename(&program_path);
-    let runtime_config = app_path::program_config(&program_name);
```
[the remark this answers is not in the document]
Comment (Ichmed) [reply podman-pilot/src/main.rs:41]: Oh completely forgot about that, i saw that in the docs but could not see how it was implemented in the code. The old code read in all the yamls and then only ever accessed the first one, am I missing something?

[the remark this answers is not in the document]
Comment (Ichmed) [reply podman-pilot/src/main.rs:41]: Oh, it appended the dicts to each other so the keys get overridden, now i get it. I thought it built a Vec with one Yaml after the other, I'll try the same thing with serde_yaml.

**Comment by Ichmed** [comment]
Re-added yaml unification and added a unit test for it.
Had to use yaml-rust since serde_yaml does not support identical keys since version 0.9. We should probably still figure out a better solution for unification but that is out of the scope of this PR

File: podman-pilot/src/config.rs [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+/// Will initialize the config on first call and return the cached version afterwards
+pub fn config() -> &'static Config<'static> {
+    &CONFIG
+}
+
+fn get_base_path() -> PathBuf {
+    which::which(env::args().next().expect("Arg 0 must be present")).expect("Symlink should exist")
+}
+
+fn load_config() -> Config<'static> {
```
Comment (schaefi) [inline podman-pilot/src/config.rs:22]: Would you mind to keep the comment that explains the reading and structure of our config files in this method ? I'm referencing the comment that you can see in the deleted ```program_config``` method ?

Would be great, thanks

**Review by schaefi — APPROVED** [review]
Looks good now, thanks for your effort :+1:

**Pull request merged**